irohad/multi_sig_transactions/state/impl/mst_state.cpp

```diff
@@ -44,30 +44,24 @@ namespace iroha {
   }
 
   bool MstState::operator==(const MstState &rhs) const {
-    const auto &lhs_batches = getBatches();
-    const auto &rhs_batches = rhs.getBatches();
-
-    return std::equal(lhs_batches.begin(),
-                      lhs_batches.end(),
-                      rhs_batches.begin(),
-                      rhs_batches.end(),
-                      [](const auto &l, const auto &r) { return *l == *r; });
+    return std::find_if(internal_state_.begin(),
+                        internal_state_.end(),
+                        [&rhs](auto &i) {
```

Comment:
Please, give more meaningful name for `i`. `mst_state_entry` or just `batch`, I think, would be fine

Reply:
It's one-liner lambda so it seems fine to use it like that